core/record/_recordValidator/attributeKeyValidator.js

```diff
@@ -19,19 +18,13 @@ const validateAttributeKey = (survey, record, attributeDef) => async (propName,
 
 const _isEntityDuplicate = (survey, record, entity) => {
   // 1. get sibling entities
-  const parentNode = Record.getParentNode(entity)(record)
-  const siblingEntitiesAndSelf = Record.getNodeChildrenByDefUuid(parentNode, Node.getNodeDefUuid(entity))(record)
-  const siblingEntities = R.reject(R.propEq(Node.keys.uuid, Node.getUuid(entity)), siblingEntitiesAndSelf)
+  const siblingEntities = Record.getNodeSiblings(entity)(record)
```

Comment:
does this return all siblings or only sibling entities?

Reply:
the node passed to getNodeSiblings is an entity -> its siblings are entities too

Reply:
why? a tree, inside a plot, cannot a have sibling like plot no? plot location?

Reply:
these are not siblings but children...

Reply:
you are right, I'll add a filter function then